### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by shonfeder**
As per our conversation today, this PR adds a [CODEOWNERS file](https://docs.github.com/en/enterprise-cloud@latest/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to our github
configuration. This provides a mechanism whereby anyone can volunteer to curate
some subset of the sourcecode. Curating some code just means you'll be tagged
for reviews of that code automatically when PRs are opened introducing changes
to it.

Anyone from the @informalsystems/apalache team who wants to should feel 
free to add themselves to the parts of the code they want to care for on 
this branch. Of course, then can also add themselves in followup changes 
as needed too.

I also added myself as a curator for the stuff that seemed most obvious to me 
given the focus of my work so far, but this is mostly meant as an example 
of the syntax.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Add your name after a .gitignore-style pattern bellow to
+# volunteer to help curate the code matching the pattern.
+
+# For syntax and configuration understood by GitHub, see
+# https://docs.github.com/en/enterprise-cloud@latest/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+
+# General application code
+
+/tla-types/ @konnov @shonfeder
```
Comment (Kukovec) [inline .github/CODEOWNERS:13]:
```suggestion
/tla-types/ @konnov @shonfeder @Kukovec
```

**Review by Kukovec — COMMENTED** [review]
I've added myself via suggestions.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# General application code
+
+/tla-types/ @konnov @shonfeder @Kukovec
+/tla-io/ @Kukovec @shonfeder
+/mod-tool/ @shonfeder
+/tla-assignments/ @Kukovec
+/tla-pp/ @Kukovec
+/tlair/ @Kukovec
+
```
Comment (rodrigo7491) [inline .github/CODEOWNERS:19]:
```suggestion
/tla-bmcmt/ @rodrigo7491
```

**Comment by shonfeder** [comment]
I’m gonna set this for merge now. Anyone else can add themselves as they wish in follow up prs.

**Review by thpani — APPROVED** [review]
@shonfeder Does @apalache-bot keep merging if nobody approved this yet? :rofl:

**Pull request merged**